### PR TITLE
Add media files duration, width, and height

### PR DESCRIPTION
**Pull request by sGerli**
Currently directus doesn't extract media files duration, width, and height.

I added ffprobe to the api project and with it and creating a temporary file it's able to get that info from the media file.
I'm currently using `'video/mpeg', 'audio/mpeg', 'audio/mp3', 'video/mp4', 'audio/ogg', 'video/ogg'` as media mime types, but probably more could be added.

ref: #4420

### CODE REVIEW

File: api/src/services/files.ts [hunk 1]
```diff
@@ -112,6 +115,27 @@ export class FilesService extends ItemsService {
 			}
 		}
 
+		if (['video/mpeg', 'audio/mpeg', 'audio/mp3', 'video/mp4', 'audio/ogg', 'video/ogg'].includes(payload.type)) {
+			const ffprobeStatic = require('@ffprobe-installer/ffprobe');
+			const buffer = await storage.disk(data.storage).getBuffer(payload.filename_disk);
+			const { path, cleanup } = await file();
+			fs.appendFile(path, buffer.content);
```
Comment (rijkvanzanten) [inline api/src/services/files.ts:122]: This'll read the whole file into memory, which might cause problems on high-usage and/or low-memory environments. Could we save the file to tmp using streams instead? (or read it directly from the stream).

From a quick google, I found another project that passes a stream to execa as input for ffprobe, maybe that could be the solution? https://github.com/caffco/get-audio-duration/blob/master/src/index.ts

Comment (sGerli) [reply api/src/services/files.ts]: Yeah, you're right I changed it and it's much faster, but now the duration only works for video files. I'll try to get it back working but if anyone knows anything about the issue please let me know.

Comment (sGerli) [reply api/src/services/files.ts]: It seems like it's not possible for audio files using stdin https://trac.ffmpeg.org/ticket/4358

Maybe we could have two versions, one for video files using streams and one for audio files using tmp files?

Comment (rijkvanzanten) [reply api/src/services/files.ts]: We could save the file to a temp location in a stream first, and then run ffmpeg. As long as we don't read the full file into Node's memory, we should be ok 👍🏻

Comment (sGerli) [reply api/src/services/files.ts]: The current version is now creating the temp file using streams and seems to be working for audio and videos.

**Comment by Nisthar** [comment]
I think this is a must have feature for storing videos.

**Comment by sGerli** [comment]
> I think this is a must have feature for storing videos.

@Nisthar yeah I've been using it to host a video an podcast app, an it's been working fine. I try to update my branch every few versions to be able to update the version of Dirextus I'm using.

**Review by br41nslug — COMMENTED** [review]
The implementation of ffprobe works great in my tests on windows and linux. The @ffprobe-installer/ffprobe library does mention that there are known permission issue with AWS which is something we should test and document.

**Comment by sGerli** [comment]
@br41nslug yeah that seems worth taking a look at. I've tested it in an EC2 but using docker, I haven't tried without docker.

**Comment by benhaynes** [comment]
Thanks for the review, @br41nslug! Not sure if any of the AWS stuff is a blocker, but we'll have @rijkvanzanten review this PR when he's back next week and get it merged if there are no outstanding issues. 👍

File: api/src/services/files.ts [hunk 2]
```diff
@@ -91,6 +97,28 @@ export class FilesService extends ItemsService {
 			payload.metadata ??= metadata;
 		}
 
+		if (['video/mpeg', 'audio/mpeg', 'audio/mp3', 'video/mp4', 'audio/ogg', 'video/ogg'].includes(payload.type)) {
```
[the remark this answers is not in the document]
Comment (sGerli) [reply api/src/services/files.ts]: Added those and the ones for wav audio

**Comment by DracoBlue** [comment]
Hello!

I was searching for video support in directus and I think this would be nice to have it added!

Best regards

**Comment by rijkvanzanten** [comment]
Heya, thanks for your PR! Our small team has been attempting to work through a backlog of external contributions for the better part of a year, but in that time many have become too stale to merge or the change is no longer desired/relevant.. 
We've updated the [contributing guidelines](https://github.com/directus/directus/blob/main/contributing.md#pull-requests) and are working on [making more resources available](https://github.com/directus/directus/discussions/17977) to make sure that doesn’t happen again 😬 In the meantime, we'll have to close this PR for now..

**Pull request closed without merging**